dask/tests/test_order.py

```diff
@@ -109,22 +109,19 @@ def test_base_of_reduce_preferred(abcde):
     assert o[(b, 1)] <= 3
 
 
-@pytest.mark.xfail(reason="Can't please 'em all")
 def test_avoid_upwards_branching(abcde):
     r"""
-         a1
-         |
-         a2
-         |
-         a3    d1
-        /  \  /
-      b1    c1
-      |     |
-      b2    c2
-            |
-            c3
-
-    Prefer b1 over c1 because it won't stick around waiting for d1 to complete
```

Comment:
I believe this explanation is actually wrong. I think `c3` should be preferred to run before `b2` because `c3->a1` is the longest critical path. 

I'm not even entirely convinced that `d1` should run before `b2`. I believe the memory pressure is identical and `b2` has a longer way to walk. Anyway, this is how it currently works and I think that's fine